app/grandchallenge/notifications/views.py

```diff
@@ -1,28 +1,61 @@
 from actstream.models import Follow
 from django.contrib import messages
 from django.contrib.messages.views import SuccessMessageMixin
-from django.http import (
-    HttpResponseNotFound,
-    HttpResponseRedirect,
-)
-from django.template.defaultfilters import pluralize
 from django.views.generic import CreateView, DeleteView, ListView
 from guardian.mixins import (
     LoginRequiredMixin,
     PermissionListMixin,
     PermissionRequiredMixin as ObjectPermissionRequiredMixin,
 )
-from guardian.shortcuts import get_objects_for_user
+from rest_framework import status, viewsets
+from rest_framework.permissions import DjangoObjectPermissions
+from rest_framework.response import Response
+from rest_framework_guardian.filters import ObjectPermissionsFilter
 
 from grandchallenge.notifications.forms import FollowForm
 from grandchallenge.notifications.models import Notification
+from grandchallenge.notifications.serializers import NotificationSerializer
 from grandchallenge.notifications.utils import (
     prefetch_generic_foreign_key_objects,
     prefetch_nested_generic_foreign_key_objects,
 )
 from grandchallenge.subdomains.utils import reverse
 
 
+class NotificationViewSet(viewsets.ModelViewSet):
```

Comment:
`ModelViewSet` includes a bunch of methods we do not use, or test. I would prefer it if we were explicit about the allowed methods, just in case the user somehow is assigned permission. If you look at https://github.com/encode/django-rest-framework/blob/e92016ac2e926483e05e296558fc3d1ea3279625/rest_framework/viewsets.py#L235-L240 it includes:

```
class ModelViewSet(mixins.CreateModelMixin,
                   mixins.RetrieveModelMixin,
                   mixins.UpdateModelMixin,
                   mixins.DestroyModelMixin,
                   mixins.ListModelMixin,
                   GenericViewSet):
```

I think that what we would need everything except the create, and we can use read only as the base as this is safer:

```
from rest_framework import mixins
...
NotificationViewSet(mixins.UpdateModelMixin,
                   mixins.DestroyModelMixin,
                   viewsets.ReadOnlyModelViewSet):
```